### PR TITLE
split signature verification from verifier

### PR DESCRIPTION
**Pull request by gerardsn**
the `Verifier` interface contains the `Verify`, `Validate`, and `VerifyVP` methods. The naming of these methods is ambiguous and there seems to be no clear separation of responsibilities. 

I've split/renamed the methods to
-  `Verify`/`VerifyVP` methods on the `Verifier` that should check the full correctness of the VC/VP
- `VerifySignature`/`verifyVPSignature` that only check if the signature on a VC/VP was valid at the given time.
where the later are now on their own struct. This should make it easier to reuse in other places

### CODE REVIEW

File: vcr/verifier/signature_verifier_test.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+	"github.com/nuts-foundation/nuts-node/crypto/storage/spi"
+	"github.com/nuts-foundation/nuts-node/jsonld"
+	"github.com/nuts-foundation/nuts-node/vdr/didjwk"
+	"github.com/nuts-foundation/nuts-node/vdr/resolver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestSignatureVerifier_VerifySignature(t *testing.T) {
```
Comment (gerardsn) [inline vcr/verifier/signature_verifier_test.go:31]: copied from verifier_test.go with minor adaptations

File: vcr/verifier/verifier_test.go [hunk 2]
```diff
@@ -558,31 +333,32 @@ func TestVerifier_VerifyVP(t *testing.T) {
...
-			require.NoError(t, err)
-			ctx := newMockContext(t)
-			ctx.keyResolver.EXPECT().ResolveKeyByID(keyID, gomock.Any(), resolver.NutsSigningKeyType).Return(publicKey, nil)
-
-			vcs, err := ctx.verifier.VerifyVP(*presentation, false, false, nil)
 
-			assert.EqualError(t, err, "verification method is not of issuer")
-			assert.Empty(t, vcs)
-		})
+		//t.Run("VP signer != VC credentialSubject.id", func(t *testing.T) {
```
Comment (gerardsn) [inline vcr/verifier/verifier_test.go:337]: @reinkrul credentialSubject.id is no longer used to derive the signer of a VP. What should happen to this test? also see #2684

**Comment by reinkrul** [comment]
Is it only a refactor, no functional changes?

File: vcr/verifier/signature_verifier.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		return sv.jsonldProof(credentialToVerify, credentialToVerify.Issuer.String(), validateAt)
+	case issuer.JWTCredentialFormat:
+		return sv.jwtSignature(credentialToVerify.Raw(), credentialToVerify.Issuer.String(), validateAt)
+	default:
+		return errors.New("unsupported credential proof format")
+	}
+}
+
+// verifyVPSignature implements the Proof Verification Algorithm: https://w3c-ccg.github.io/data-integrity-spec/#proof-verification-algorithm
+func (sv *signatureVerifier) verifyVPSignature(presentation vc.VerifiablePresentation, validateAt *time.Time) error {
```
Comment (reinkrul) [inline vcr/verifier/signature_verifier.go:40]: that this one unexported, and the VerifySignature exported is weird

File: vcr/verifier/signature_verifier.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	}
+	return nil
+}
+
+func (sv *signatureVerifier) jwtSignature(jwtDocumentToVerify string, issuer string, at *time.Time) error {
+	var keyID string
+	_, err := crypto.ParseJWT(jwtDocumentToVerify, func(kid string) (crypt.PublicKey, error) {
+		keyID = kid
+		return sv.resolveSigningKey(kid, issuer, at)
+	}, jwt.WithClock(jwt.ClockFunc(func() time.Time {
```
Comment (reinkrul) [inline vcr/verifier/signature_verifier.go:103]: I realized clock skew is missing here..

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: vcr/verifier/verifier_test.go (hunk 2 above)
Comment (gerardsn) [reply vcr/verifier/verifier_test.go]: added check to VerifyVP and re-enabled test

File: vcr/verifier/signature_verifier.go (hunk 3 above)
Comment (gerardsn) [reply vcr/verifier/signature_verifier.go]: exported; the signatureVerifier should probably get its own interface, but out of scope for now

File: vcr/verifier/signature_verifier.go (hunk 4 above)
Comment (gerardsn) [reply vcr/verifier/signature_verifier.go:103]: deferred to #2684

**Review by reinkrul — APPROVED** [review]

**Pull request merged**